module/zfs/arc.c

```diff
@@ -311,6 +311,11 @@
 boolean_t arc_watch = B_FALSE;
 #endif
 
+/* Dirty data threshold if not overridden by the module parameter */
+#define ZFS_DIRTY_MAX_LIMIT		(4ULL * 1024 * 1024 * 1024)
+
+uint64_t		allmem;
```

Comment:
I'd suggest moving this in to the arcstats if we're effectively going to cache it.  That way there's some visibility it to what values the ARC of operating with.